src/mc/nucleosome.f03

```diff
@@ -81,19 +83,127 @@ function nucleosome_energy(RP1,R,UP1,U,VP1,V,linkBP,wrapBP)
 
     call nucleosomeProp(U,V,R,linkBP,wrapBP,Utemp,Vtemp,Rtemp)
 
+    !  need to interpolate between basepairs if fractional
+    call get_params(linkBP,EB,EPAR,EPERP,GAM,ETA,XIR,XIU,sigma,etwist,simtype)
+
     nucleosome_energy =  E_SSWLCWT(RP1,Rtemp,UP1,Utemp,VP1,Vtemp, &
-                   multiParams(1,linkBP),&   ! EB
-                   multiParams(2,linkBP),&   ! EPAR
-                   multiParams(3,linkBP),&   ! EPERP
-                   multiParams(5,linkBP),&   ! ETA
-                   multiParams(4,linkBP),&   ! GAM
-                   multiParams(9,linkBP))   ! etwist
+                   EB,&!multiParams(1,linkBP),&   ! EB
+                   EPAR,&!multiParams(2,linkBP),&   ! EPAR
+                   EPERP,&!multiParams(3,linkBP),&   ! EPERP
+                   ETA,&!multiParams(5,linkBP),&   ! ETA
+                   GAM,&!multiParams(4,linkBP),&   ! GAM
+                   ETWIST)!multiParams(9,linkBP))   ! etwist
 end function nucleosome_energy
 
+!  ------------------------------------------------------
+!
+! internucleosomal energy (i.e. faces attracted via harmonic
+! oscillator), need spring constant and preferred distance
+!
+!  ------------------------------------------------------
+function internucleosome_energy(RI,RJ,UI,UJ,VI,VJ)
```

Comment:
In the description could you make it clear that this is the energy of the DNA linker rather than the energy an internucleosome interaction energy.  Also, what exactly does this do different the the normal SSWLCWT?

Reply:
wait no, sorry for the confusion. this is supposed to be the energy of an internucleosome interaction energy. it uses the latest work from de pablo's group where they differentiate the attraction energies of histone-histone, histone-DNA, and DNA-DNA between two nucleosomes. i just tried to coarse-grain that energy here. we can talk more about this later if you want. but in the meantime, i will definitely add better documentation.